src/main/java/org/jenkinsci/plugins/workflow/support/PipelineIOUtils.java

```diff
@@ -23,11 +25,36 @@
      * @param xstream xstream to use for output
      * @param atomicWrite If true, do an atomic write, otherwise do a direct write to file.
      * @throws IOException
+     * @deprecated use {@link #writeByXStream(Object, File, XStream2, boolean, boolean)}
      */
+    @Deprecated
     public static void writeByXStream(@Nonnull Object toWrite, @Nonnull File location, @Nonnull XStream2 xstream, boolean atomicWrite) throws IOException {
+        writeByXStream(toWrite, location, xstream, atomicWrite, atomicWrite);
+    }
+
+    /**
+     * Convenience method to transparently write data directly or atomically using {@link
+     * hudson.util.AtomicFileWriter}, with or without {@link FileChannel#force} (i.e., {@code
+     * fsync} or {@code FlushFileBuffers}).
+     *
+     * @param toWrite Object to write to file
+     * @param location File to write object to
+     * @param xstream xstream to use for output
+     * @param atomicWrite use {@link AtomicFileWriter} to write the file.
+     * @param force If true, call {@link FileChannel#force} (i.e., {@code fsync} or {@code
+     *     FlushFileBuffers}) after writing the file.
+     * @throws IOException
+     */
+    public static void writeByXStream(
+            @Nonnull Object toWrite,
+            @Nonnull File location,
+            @Nonnull XStream2 xstream,
+            boolean atomicWrite,
+            boolean force)
+            throws IOException {
         if (atomicWrite) {
             XmlFile file = new XmlFile(xstream, location);
-            file.write(toWrite);
+            file.write(toWrite, force);
```

Comment:
Is there any way to do this such that the plugin doesn't have to be limited to Jenkins versions with the upstream change?  If not, we'll have to maintain split release trains, which isn't terrible but is added overhead.

Reply:
Not that I'm aware of, I'm afraid.

Reply:
@basil 
Excuse me, what I mean is using reflection such as in https://github.com/jenkinsci/scm-api-plugin/commit/783b9824f567ec1cbccc91f601064a932af59636 

Or would that result in horrible performance?

Reply:
Not sure about performance, but using reflection to avoid core API bumps has historically caused a number of problems in Pipeline, including jenkinsci/workflow-basic-steps-plugin#127. In general I think Jesse's preference has historically been to keep things simple and avoid reflection for that reason.

Reply:
Yup. If and when you actually come across a bug fix that you wish to make available to users of LTS or even older versions, https://gist.github.com/jglick/86a30894446ed38f918050c1180483e2 is easy enough. Not worth making life complicated “just in case” that happens.